File: backend/services/greenProductsService.py
import json
from pathlib import Path
from typing import List, Dict

PROJECT_ROOT = Path(__file__).resolve().parent.parent.parent
DATA_PATH = PROJECT_ROOT / "backend" / "green_products_mock_data.json"


def load_green_products_data() -> Dict:
    path = DATA_PATH
    if not path.exists():
        path = Path(__file__).resolve().parent.parent / "green_products_mock_data.json"
    with path.open("r", encoding="utf-8") as fh:
        return json.load(fh)


def get_available_green_products() -> List[Dict]:
    catalog = load_green_products_data()
    return [product for product in (catalog.get("products") or []) if product.get("status") == "ACTIVE"]

File: backend/greenProductsService.js
const fs = require('fs');
const path = require('path');

const dataPath = path.join(__dirname, 'green_products_mock_data.json');

function loadGreenProductsData() {
  const rawData = fs.readFileSync(dataPath, 'utf8');
  return JSON.parse(rawData);
}

function getAvailableGreenProducts() {
  const catalog = loadGreenProductsData();
  return (catalog.products || []).filter((product) => product.status === 'ACTIVE');
}

module.exports = {
  getAvailableGreenProducts,
  loadGreenProductsData,
};
